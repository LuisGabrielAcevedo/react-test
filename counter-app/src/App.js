import React from "react";
import PropTypes from "prop-types";
import "./App.css";

const App = ({ saludo, subtitulo }) => {
  return (
    <>
      <h1>{saludo}</h1>
      <p>{subtitulo}</p>
    </>
  );
};

export default App;

App.propTypes = {
  saludo: PropTypes.string.isRequired,
};

App.defaultProps = {
  subtitulo: "Soy un subtitulo",
};

// Print Json
// const obj = {
//   nombre: "Luis",
//   edad: 29,
// };
// <pre>{JSON.stringify(obj, null, 3)}</pre>
