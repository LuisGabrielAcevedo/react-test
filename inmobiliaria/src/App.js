import React, { useEffect } from "react";
import "./App.css";
import PropertyList from "./components/views/property-list";
import NavBar from "./components/layout/nav-bar";
import theme from "./theme/theme";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { Grid, MuiThemeProvider } from "./material-ui";
import Register from "./components/security/register";
import Login from "./components/security/login";
import { FirebaseContext } from "./server";

function App(props) {
  let firebase = React.useContext(FirebaseContext);
  const [logged, setLogged] = React.useState(false);

  useEffect(() => {
    firebase.isLogged().then((resp) => {
      setLogged(true);
    })
  });

  return logged  ? (
    <Router>
      <MuiThemeProvider theme={theme}>
        <NavBar />
        <Grid container>
          <Switch>
            <Route path="/" exact component={PropertyList} />
            <Route path="/auth/register" exact component={Register} />
            <Route path="/auth/login" exact component={Login} />
          </Switch>
        </Grid>
      </MuiThemeProvider>
    </Router>
  ): null;
}

export default App;
