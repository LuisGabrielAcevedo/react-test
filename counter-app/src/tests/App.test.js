import "@testing-library/jest-dom";
import App from "../App";
import React from "react";
import { shallow } from "enzyme";

describe("Pruebas en componente App.js", () => {
  //   Cuando se usa jest que viene por defecto en react
  //   test("Debe de mostrar el mensaje: Hola, soy gabriel", () => {
  //     const saludo = "Hola, soy gabriel";
  //     const { getByText } = render(<App saludo={saludo} />);
  //     expect(getByText(saludo)).toBeInTheDocument();
  //   });
  test("debe de mostrar <App/> correctamente", () => {
    const saludo = "Hola, soy gabriel";
    const wrapper = shallow(<App saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola, soy gabriel";
    const subtitulo = "Soy un subtitulo";
    const wrapper = shallow(<App saludo={saludo} subtitulo={subtitulo} />);
    const textoParrafo = wrapper.find("p").text();
    expect(textoParrafo).toBe(subtitulo);
  });
});
