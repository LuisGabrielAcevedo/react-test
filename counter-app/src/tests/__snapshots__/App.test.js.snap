// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente App.js debe de mostrar <App/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy gabriel
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en componente App.js debe de mostrar el titulo enviado por props 1`] = `
<Fragment>
  <h1>
    Hola, soy gabriel
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
